tests/test_configure.py

```diff
@@ -94,3 +173,212 @@ def test_set_up_configuration_missing_csv():
+]
+
+
+def check_credential_deletion(config_file, removed_index):
+    """Check that a particular credential is not in a config file after removal,
+       and that all other are still present.
+
+    Args:
+        config_file: Path to the file to check.
+        removed_index: Index of the credential in delete_input_creds which should not be present.
+    """
+    observed = auth.load_auth(config_file)
+    for idx, cred in enumerate(delete_input_creds):
+        if idx == removed_index:
+            assert cred not in observed
+        else:
+            assert cred in observed
+
+
+def test_delete_credential_no_match(tmpdir):
+    invalid_username = "not-a-user"
+    with temp_environ(HOME=str(tmpdir)), \
+            patch("hop.auth.load_auth", MagicMock(return_value=delete_input_creds.copy())), \
+            pytest.raises(RuntimeError) as err:
+        configure.delete_credential(invalid_username)
+    assert "No matching credential found" in err.value.args[0]
+    assert invalid_username in err.value.args[0]
+
+    invalid_host = "example.com"
+    with temp_environ(HOME=str(tmpdir)), \
+            patch("hop.auth.load_auth", MagicMock(return_value=delete_input_creds.copy())), \
+            pytest.raises(RuntimeError) as err:
+        configure.delete_credential(f"{invalid_username}@{invalid_host}")
+    assert "No matching credential found" in err.value.args[0]
+    assert invalid_username in err.value.args[0]
+    assert f" with hostname '{invalid_host}'" in err.value.args[0]
+
+
+def test_delete_credential_no_match_no_data(tmpdir):
+    invalid_username = "not-a-user"
+    with temp_environ(HOME=str(tmpdir)), \
+            pytest.raises(RuntimeError) as err:
+        configure.delete_credential(invalid_username)
+    assert "No matching credential found" in err.value.args[0]
+    assert invalid_username in err.value.args[0]
+
+
+def test_delete_credential_no_hostname(tmpdir):
+    with temp_environ(HOME=str(tmpdir)):
+        with patch("hop.auth.load_auth", MagicMock(return_value=delete_input_creds.copy())):
+            configure.delete_credential("user1")
+        check_credential_deletion(configure.get_config_path("auth"), 0)
+
+
+def test_delete_credential_with_hostname(tmpdir):
+    with temp_environ(HOME=str(tmpdir)):
+        with patch("hop.auth.load_auth", MagicMock(return_value=delete_input_creds.copy())):
+            configure.delete_credential("user3@example.com")
+        check_credential_deletion(configure.get_config_path("auth"), 2)
+
+
+def test_delete_credential_ambiguous(tmpdir):
+    with temp_environ(HOME=str(tmpdir)), \
+            patch("hop.auth.load_auth", MagicMock(return_value=delete_input_creds.copy())), \
+            pytest.raises(RuntimeError) as err:
+        configure.delete_credential("user3")
+    assert "Ambiguous credentials found" in err.value.args[0]
+    assert delete_input_creds[2].username in err.value.args[0]
+    assert delete_input_creds[2].hostname in err.value.args[0]
```

Comment:
The fist line checks the username, while the second checks the hostname. There is an asymmetry because `delete_input_creds[3].username` should in principle be checked, but since the test is one of an ambiguous username this would not (with the simple `in` test being used) check anything distinct from the username check for credential 2. Since credentials 0 and 1 have different usernames, they should not appear in the error message being checked.

Reply:
Oh I see, I clearly can't read. Thanks, that makes sense.